util/predicates/generic_predicates.go

```diff
@@ -303,3 +308,24 @@ func processIfTopologyOwned(scheme *runtime.Scheme, logger logr.Logger, obj clie
 	logger.V(6).Info("Resource is not topology owned, will not attempt to map resource")
 	return false
 }
+
+// ResourceIsUnchanged returns a predicate that returns true only if the resource
+// has changed. This predicate allows to drop events which come resync events on
+// additionally watched objects.
+func ResourceIsUnchanged() predicate.Funcs {
+	return TypedResourceIsUnchanged[client.Object]()
+}
+
+// TypedResourceIsUnchanged returns a predicate that returns true only if the resource
+// has changed. This predicate allows to drop events which come resync events on
+// additionally watched objects.
+func TypedResourceIsUnchanged[T client.Object]() predicate.TypedFuncs[T] {
+	return predicate.TypedFuncs[T]{
+		UpdateFunc: func(e event.TypedUpdateEvent[T]) bool {
+			return e.ObjectOld.GetResourceVersion() != e.ObjectNew.GetResourceVersion()
```

Comment:
Question: @sbueringer / @fabriziopandini : should we log something?

Reply:
Yup log level 6 please